pyannote/core/annotation.py

```diff
@@ -1416,8 +1419,11 @@ def discretize(
             )
 
         start_frame = resolution.closest_frame(start_time)
-        end_frame = resolution.closest_frame(end_time)
-        num_frames = end_frame - start_frame
+        if fixed:
+            num_frames = int(round(support.duration / resolution.step))
```

Comment:
You might end up with the same problem that you are trying to solve for some combination of `support.duration` and `resolution.step` as `support.duration` is computed as `support.end - support.start`. 

Providing the `duration` directly will solve that problem.

Reply:
```suggestion
            num_frames = int(round(duration / resolution.step))
```